crates/lib/src/qpu/translation.rs

```diff
@@ -128,16 +140,109 @@ impl TranslationOptions {
         self.inner.translation_backend.as_ref()
     }
 
+    /// Get a mutable reference to the backend used for translation.
+    #[must_use]
+    pub fn backend_mut(&mut self) -> Option<&mut TranslationBackend> {
+        self.inner.translation_backend.as_mut()
+    }
+
     /// Use the first-generation translation backend available on QCS since 2018.
-    pub fn use_backend_v1(&mut self) {
-        self.inner.translation_backend =
-            Some(TranslationBackend::V1(translation::BackendV1Options {}));
+    pub fn with_backend_v1(&mut self) -> &mut BackendV1Options {
+        let backend = &mut self.inner.translation_backend;
+        if let Some(TranslationBackend::V1(options)) = backend {
+            return options;
+        }
+
+        *backend = Some(TranslationBackend::V1(BackendV1Options::default()));
+        let Some(TranslationBackend::V1(options)) = backend.as_mut() else {
+            unreachable!("backend was just set to V1")
+        };
+        options
     }
 
     /// Use the second-generation translation backend available on QCS since 2023
-    pub fn use_backend_v2(&mut self) {
-        self.inner.translation_backend =
-            Some(TranslationBackend::V2(translation::BackendV2Options {}));
+    pub fn with_backend_v2(&mut self) -> &mut BackendV2Options {
+        let backend = &mut self.inner.translation_backend;
+        if let Some(TranslationBackend::V2(options)) = backend {
+            return options;
+        }
+
+        *backend = Some(TranslationBackend::V2(BackendV2Options::default()));
+        let Some(TranslationBackend::V2(options)) = backend.as_mut() else {
+            unreachable!("backend was just set to V2")
+        };
+        options
+    }
+
+    fn ensure_backend_v1(&mut self) -> Result<&mut BackendV1Options, TranslationBackendMismatch> {
+        if matches!(self.backend(), None | Some(TranslationBackend::V1(_))) {
+            Ok(self.with_backend_v1())
+        } else {
+            Err(TranslationBackendMismatch::V1)
+        }
+    }
+
+    fn ensure_backend_v2(&mut self) -> Result<&mut BackendV2Options, TranslationBackendMismatch> {
+        if matches!(self.backend(), None | Some(TranslationBackend::V2(_))) {
+            Ok(self.with_backend_v2())
+        } else {
+            Err(TranslationBackendMismatch::V2)
+        }
+    }
+
+    /// If `false`, default calibrations will not be prepended to the translated program.
+    ///
+    /// # Errors
+    ///
+    /// This will return an error if the translation backend is set to something other than V2.
+    pub fn v2_prepend_default_calibrations(
```

Comment:
I wonder if we should just drop the `v2` in the name here since `v2` is eventually going to be the only option.

Reply:
Fair point. My thought was having the v2 in the name makes it more clear which backend must be set to avoid an error when calling the method, since it is only valid for v2.

We could also discuss whether the error makes sense -- I think it does, since the alternatives of silently replacing the backend version or doing nothing on mismatch seems like bad UX.

Reply:
I think returning an error is a good choice for sure since it avoids any surprises. I think it makes dropping `v2` a valid option, since in the worst case, users will miss the docstring and get an error guiding them to use backend v2. In any case, not something that will keep me from letting this ship, so I'll leave it to you to decide.